error: use of bitwise '|' with boolean operands

```
disk-samsung/freebsd-ports/devel/tket/work/tket-1.4.3/tket/src/Architecture/Architecture.cpp:188:11: error: use of bitwise '|' with boolean operands [-Werror,-Wbitwise-instead-of-logical]
          edge_exists(Node(i), Node(j)) | edge_exists(Node(j), Node(i));
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                                        ||
/disk-samsung/freebsd-ports/devel/tket/work/tket-1.4.3/tket/src/Architecture/Architecture.cpp:188:11: note: cast one or both operands to int to silence this warning
1 error generated.
```

Version: 1.4.3
clang-14
FreeBSD 13.1

--- a/tket/src/ArchAwareSynth/Path.cpp
+++ b/tket/src/ArchAwareSynth/Path.cpp
@@ -110,7 +110,7 @@ PathHandler PathHandler::construct_acyclic_handler() const {
   while (!current_layer_vertices.empty()) {
     for (unsigned vert : current_layer_vertices) {
       for (unsigned j = 0; j != n; ++j) {
-        if ((distance_matrix_(vert, j) == 1) & (!vertices_in_tree[j])) {
+        if ((distance_matrix_(vert, j) == 1) && (!vertices_in_tree[j])) {
           // if first encounter of vertex, add to next layer
           if (parents_neighbours[j] == empty_pair) {
             next_layer_vertices.push_back(j);

--- a/tket/src/Architecture/Architecture.cpp
+++ b/tket/src/Architecture/Architecture.cpp
@@ -185,7 +185,7 @@ MatrixXb Architecture::get_connectivity() const {
   for (unsigned i = 0; i != n; ++i) {
     for (unsigned j = 0; j != n; ++j) {
       connectivity(i, j) =
-          edge_exists(Node(i), Node(j)) | edge_exists(Node(j), Node(i));
+          edge_exists(Node(i), Node(j)) || edge_exists(Node(j), Node(i));
     }
   }
   return connectivity;

